rust-src/bulletproofs/src/inner_product_proof.rs

```diff
@@ -331,6 +332,118 @@ pub fn verify_inner_product<C: Curve>(
     P_prime.minus_point(&RHS).is_zero_point()
 }
 
+/// This function is an optimized variant of the above.
+/// It is verified whether P'=<a,G>+<b,H'>+<a,b>Q for P' =
```

Comment:
It seems like this function is verifying the same as the above one, is that correct?

In that case, why do we retain the old one.

It would also be useful to include some data as to what "optimized" means. What are the gains? That can be put into the PR description.

Reply:
The old function is used in the range proof file, so we'd need to adjust that if we want to remove the old function.

Reply:
So why not if the behaviour is the same?

Reply:
It takes different inputs, so the naive range proof verification won't work. Note that the range proof also has a naive and ultra efficient variant etc., and that could also be streamlined with the set-membership proof. But seemed out of scope for this. I think ideally we'd kill the naive range proof verification and adjust the regular one to use the new inner product with scalars function.

Reply:
Is this resolved for now? We can make a separate PR for revising the range proof and killing the then unused functions.